src/bank.rs

```diff
@@ -430,6 +433,18 @@ impl Bank {
     pub fn transaction_count(&self) -> usize {
         self.transaction_count.load(Ordering::Relaxed)
     }
+
+    pub fn check_signature(&self, signature: &Signature) -> Option<(Hash, Signature)> {
+        let last_ids_sigs = self.last_ids_sigs
+            .read()
+            .expect("'last_ids_sigs' read lock");
+        for (hash, signatures) in last_ids_sigs.iter() {
+            if let Some(sig) = signatures.get(signature) {
+                return Some((*hash, *sig));
```

Comment:
Just returning a bool is fine. That hash is the `last_id` that was used in the transaction, and not the hash of the ledger entry containing the transaction. If you return a hash, folks might assume its the latter.